tools/mason/MasonInit.chpl

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2004-2020 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use TOML;
+use Path; 
+use Spawn;
+use MasonEnv;
+use MasonNew;
+use MasonBuild;
+use MasonHelp;
+use MasonUtils;
+use MasonExample;
+use MasonModify;
+use FileSystem;
+private use List;
+
+/* 
+Initialises a library project in a project directory
+  mason init <dirName/path>
+  or mason init (inside project directory)
+*/
+proc masonInit(args) throws {
+  try! {
+    var name = '';
+    for arg in args[2..]{
+      name = arg;
+    }
+    //checks if path is given as parameter or not
+    if name == '' {
+      const cwd = getEnv("PWD");
+      const name = basename(cwd);
+      const path = '.';
+      //If Mason.toml not present, proceed with init
+      validateMasonFile(path, name);
+      validateInit(path);
+      writeln("Initialized new library project: " + name);
+    } 
+    else {
+      // if the target directory in path doesnt exist, throw error
+      // if target directory exists, check for files && validate
+      // create folders and toml file without overwriting anything 
+      // if TOML file exists, check for values in it and validate
+      const path = name;
+      if isDir(path) {
+        //make a condition to handle Mason.toml & get rid from below
+        validateMasonFile(path, basename(path));
+        validateInit(path);
+        writeln("Initialized new library project in " + path + ": " + basename(path));
+      } 
+      else {
+        throw new owned MasonError("Directory does not exist:" + path +
+                              ". Did you mean 'mason new' to create a new project from scratch?"); 
```

Comment:
Can we put `Did you mean 'mason new' to create a new project from scratch?`on a new line?